modules/24.05-compat.nix

```diff
@@ -0,0 +1,6 @@
+{ lib, ... }: {
+  imports = lib.optionals (lib.versionOlder lib.version "24.11pre") [
+    (lib.mkAliasOptionModule [ "hardware" "graphics" "enable" ] [ "hardware" "opengl" "enable" ])
+    (lib.mkAliasOptionModule [ "hardware" "graphics" "extraPackages" ] [ "hardware" "opengl" "extraPackages" ])
+  ];
```

Comment:
You are sure that this is not causing more problems?

Reply:
the worst one i could think of is 24.05 users thinking `hardware.graphics` *is* present when it's not - but other than that, no

what's basically the same exact method is also used [here](https://github.com/NixOS/nixos-hardware/blob/901bc809b5d3e73a609a167385df23311d81b39c/common/gpu/24.05-compat.nix) in nixos-hardware

Reply:
Would including the mentioned approbriate hardware.nix be an option then or leaving it to the user all at once and mentioning it in the docs?